From: Lucas Stach <l.stach@pengutronix.de>
Date: Fri, 2 Aug 2019 14:53:08 +0200
Subject: etnaviv: check for softpin availability on Halti5 devices

Halti5 uses texture descriptors to control the samplers, and thus needs to
know the GPU virtual address for the texture buffers to fill into the
descriptor buffer. Without softpin userspace has no control over the GPU
VM and also no way to fix up the texture descriptor buffer, so there is
no point in creating a screen on a Halti5 device without softpin being
available.

Signed-off-by: Lucas Stach <l.stach@pengutronix.de>
---
 src/gallium/drivers/etnaviv/etnaviv_screen.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/gallium/drivers/etnaviv/etnaviv_screen.c b/src/gallium/drivers/etnaviv/etnaviv_screen.c
index c1cc56b..5a42eca 100644
--- a/src/gallium/drivers/etnaviv/etnaviv_screen.c
+++ b/src/gallium/drivers/etnaviv/etnaviv_screen.c
@@ -917,6 +917,11 @@ etna_screen_create(struct etna_device *dev, struct etna_gpu *gpu,
       .lower_sincos = !screen->specs.has_sin_cos_sqrt,
    };
 
+   if (screen->specs.halti >= 5 && !etnaviv_device_softpin_capable(dev)) {
+       DBG("halti5 requires softpin");
+      goto fail;
+   }
+
    /* apply debug options that disable individual features */
    if (DBG_ENABLED(ETNA_DBG_NO_EARLY_Z))
       screen->features[viv_chipFeatures] |= chipFeatures_NO_EARLY_Z;
